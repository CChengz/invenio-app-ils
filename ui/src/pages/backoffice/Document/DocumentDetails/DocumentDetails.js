import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Container } from 'semantic-ui-react';
import history from '@history';
import { Loader, Error } from '@components';
import {
  DocumentMetadata,
  DocumentPendingLoans,
  DocumentRelations,
  DocumentStats,
  DocumentItems,
} from './components';

export default class DocumentDetails extends Component {
  constructor(props) {
    super(props);
    this.deleteDocument = this.props.deleteDocument;
    this.fetchDocumentDetails = this.props.fetchDocumentDetails;
  }

  componentDidMount() {
    this.unlisten = history.listen(loc => {
      if (loc.state && loc.state.pid && loc.state.type === 'Document') {
        this.fetchDocumentDetails(loc.state.pid);
      }
    });
    this.fetchDocumentDetails(this.props.match.params.documentPid);
  }

  componentWillUnmount() {
    this.unlisten();
  }

  render() {
    const { isLoading, error } = this.props;
    return (
      <Container>
        <Loader isLoading={isLoading}>
          <Error error={error}>
            <DocumentMetadata />
            <DocumentPendingLoans />
            <DocumentItems />
            <DocumentRelations />
            <DocumentStats />
          </Error>
        </Loader>
      </Container>
    );
  }
}

DocumentDetails.propTypes = {
  deleteDocument: PropTypes.func.isRequired,
  fetchDocumentDetails: PropTypes.func.isRequired,
};
