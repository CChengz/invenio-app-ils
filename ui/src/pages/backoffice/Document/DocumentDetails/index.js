import { connect } from 'react-redux';

import { deleteDocument, fetchDocumentDetails } from './state/actions';
import DocumentDetailsComponent from './DocumentDetails';

const mapStateToProps = state => ({
  isLoading: state.documentDetails.isLoading,
  error: state.documentDetails.error,
  hasError: state.documentDetails.hasError,
});

const mapDispatchToProps = dispatch => ({
  fetchDocumentDetails: documentPid =>
    dispatch(fetchDocumentDetails(documentPid)),
  deleteDocument: documentPid => dispatch(deleteDocument(documentPid)),
});

export const DocumentDetails = connect(
  mapStateToProps,
  mapDispatchToProps
)(DocumentDetailsComponent);
