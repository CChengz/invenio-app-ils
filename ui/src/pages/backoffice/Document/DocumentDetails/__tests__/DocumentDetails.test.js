import React from 'react';
import { shallow, mount } from 'enzyme';
import DocumentDetails from '../DocumentDetails';

jest.mock('../components/', () => {
  return {
    DocumentMetadata: () => null,
    DocumentPendingLoans: () => null,
    DocumentItems: () => null,
    DocumentRelations: () => null,
    DocumentStats: () => null,
  };
});

describe('DocumentDetails tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  const routerHistory = {
    listen: () => () => {},
  };
  const routerUrlParams = {
    params: {
      documentPid: 111,
    },
  };

  it('should load the details component', () => {
    const component = shallow(
      <DocumentDetails
        history={routerHistory}
        match={routerUrlParams}
        fetchDocumentDetails={() => {}}
        deleteDocument={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should fetch document details on mount', () => {
    const mockedFetchDocumentDetails = jest.fn();
    const mockDeleteDocument = jest.fn();
    component = mount(
      <DocumentDetails
        history={routerHistory}
        match={routerUrlParams}
        fetchDocumentDetails={mockedFetchDocumentDetails}
        deleteDocument={mockDeleteDocument}
      />
    );
    expect(mockedFetchDocumentDetails).toHaveBeenCalledWith(
      routerUrlParams.params.documentPid
    );
  });
});
