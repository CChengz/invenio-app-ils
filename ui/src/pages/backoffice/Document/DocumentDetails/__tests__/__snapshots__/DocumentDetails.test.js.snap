// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentDetails tests should load the details component 1`] = `
<Container>
  <Loader>
    <Error>
      <DocumentMetadata />
      <DocumentPendingLoans />
      <DocumentItems />
      <DocumentRelations />
      <DocumentStats />
    </Error>
  </Loader>
</Container>
`;
